File: app/src/main/java/com/temperoni/recipes/utils/Decoder.java
package com.temperoni.recipes.utils;

import com.google.gson.Gson;
import com.temperoni.recipes.domain.dto.Recipe;

import org.apache.commons.io.IOUtils;

import java.io.IOException;
import java.util.List;

/**
 * @author Leandro Temperoni
 */
public class Decoder {

    public List<Recipe> decodeJson(String fileName) {
        return new Gson().fromJson(getJson(fileName), RecipeList.class).getRecipes();
    }

    private String getJson(String fileName) {
        String result = "";

        ClassLoader classLoader = getClass().getClassLoader();
        try {
            result = IOUtils.toString(classLoader.getResourceAsStream(fileName));
        } catch (IOException e) {
            e.printStackTrace();
        }

        return result;
    }

    private class RecipeList {
        private List<Recipe> recipes;

        public List<Recipe> getRecipes() {
            return recipes;
        }
    }
}

File: app/src/main/java/com/temperoni/recipes/utils/Decoder.kt
package com.temperoni.recipes.utils

import com.google.gson.Gson
import com.temperoni.recipes.domain.dto.Recipe
import org.apache.commons.io.IOUtils

import java.io.IOException

/**
 * @author Leandro Temperoni
 */
class Decoder {

    fun decodeJson(fileName: String): List<Recipe>? {
        return Gson().fromJson(getJson(fileName), RecipeList::class.java).recipes
    }

    private fun getJson(fileName: String): String {
        var result = ""

        val classLoader = javaClass.classLoader
        try {
            result = IOUtils.toString(classLoader?.getResourceAsStream(fileName)?.reader())
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return result
    }

    private inner class RecipeList {
        val recipes: List<Recipe>? = null
    }
}
